### PR TITLE
Add Hostname for Database Services status

### PR DESCRIPTION
**Pull request by marcoandredinis**
When heartbeating the status of a Database Service, we are now including the Hostname (as defined in `teleport.yaml/teleport.nodename`).

This helps users troubleshoot their access to a particular set of databases.

changelog: The hostname where the process is running is returned when running `tctl get db_services`.

Demo
`tctl get db_services`
```yaml
kind: db_service
metadata:
  expires: "2024-08-27T14:18:07.682498Z"
  name: e2cd22d5-db43-46ce-82b7-f65914cabe49
  revision: 8fe1d547-73dc-46af-9c71-e531a05073e9
spec:
  hostname: lenix
  resources:
  - aws: {}
    labels:
      env: prod
version: v1
```

### CODE REVIEW

File: docs/pages/enroll-resources/database-access/guides/dynamic-registration.mdx [hunk 1]
```diff
@@ -36,6 +36,26 @@ resources:
...
+      env: prod
+      engine: postgres
+  - labels:
+      env: test
+      engine: mysql
+status:
+  hostname: my-host
+version: v1
+```
+
```
Comment (stevenGravy) [inline docs/pages/enroll-resources/database-access/guides/dynamic-registration.mdx:58]:
```suggestion

The `name` within a `db_service` matches to the host ID of the agent running the Teleport Database Service.

```

**Comment by espadolini** [comment]
Why is the node name in `status`? the node name is in `spec` in `ServerV2`.

**Comment by marcoandredinis** [comment]
> Why is the node name in `status`? the node name is in `spec` in `ServerV2`.

**Comment by marcoandredinis** [comment]
> Why is the node name in `status`? the node name is in `spec` in `ServerV2`.

ServerV2 does not have a status field, which is something we don't historically add to resources.

At first, it seemed more natural to store it in the status because it's something we cant update. It represents the current state of that particular resource.

Having said that, if you think it should go to the spec, I can move it 👍 
@espadolini

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -5710,6 +5710,17 @@ message DatabaseServiceV1 {
...
   ];
+  // Status is the resource status.
+  DatabaseServiceStatusV1 Status = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "status"
+  ];
+}
+
+// DatabaseServiceStatusV1 is the DatabaseService Status.
+message DatabaseServiceStatusV1 {
```
Comment (greedy52) [inline api/proto/teleport/legacy/types/types.proto:5721]: Feels weird to me that HostID is part of metadata but Hostname is status.  Do Hostname change? Is `DatabaseServiceV1` a heartbeat or it has other uses?

Comment (greedy52) [reply api/proto/teleport/legacy/types/types.proto:5721]: anything else other than hostname that could be useful?

**Comment by espadolini** [comment]
> It represents the current state of that particular resource.

IMO either the entirety of the heartbeat is "status", or a resource's status _is_ the "spec" of the corresponding heartbeat.

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (marcoandredinis) [reply api/proto/teleport/legacy/types/types.proto:5721]: > Do Hostname change?

The hostname can change, if you change the host's hostname (unless you hardcoded it in `teleport.yaml/teleport.nodename`.

> Is DatabaseServiceV1 a heartbeat or it has other uses?

It's only a heartbeat.

> anything else other than hostname that could be useful?

I was thinking about adding the proxied databases in that particular service. Probably in another PR.

**Review by espadolini — APPROVED** [review]

**Review by stevenGravy — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (greedy52) [reply api/proto/teleport/legacy/types/types.proto]: i was thinking about the cloud identity that the host uses, e.g. AWS caller identity. WDYT? cc @GavinFrazar

Comment (marcoandredinis) [reply api/proto/teleport/legacy/types/types.proto]: That sounds helpful as well 👍

**Pull request merged**